pkg/grpc/tls_client_certificate_authenticator.go

```diff
@@ -61,5 +65,49 @@ func (a *tlsClientCertificateAuthenticator) Authenticate(ctx context.Context) (*
 	if _, err := certs[0].Verify(opts); err != nil {
 		return nil, util.StatusWrapWithCode(err, codes.Unauthenticated, "Cannot validate TLS client certificate")
 	}
-	return a.metadata, nil
+
+	searchContext, err := getClientCertificateJMESPathSearchContext(certs[0])
+	if err != nil {
+		return nil, util.StatusWrapWithCode(err, codes.Unauthenticated, "Failed to create a search context for extracting metadata")
+	}
+
+	// Validate the client cert matches our expectations
```

Comment:
Please end all sentences in comments with a full stop (period), also in the Protobuf source files.